lxd/storage/drivers/driver_powerflex_volumes.go

```diff
@@ -0,0 +1,1072 @@
+package drivers
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"os"
+	"strings"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/canonical/lxd/lxd/backup"
+	deviceConfig "github.com/canonical/lxd/lxd/device/config"
+	"github.com/canonical/lxd/lxd/instancewriter"
+	"github.com/canonical/lxd/lxd/migration"
+	"github.com/canonical/lxd/lxd/operations"
+	"github.com/canonical/lxd/lxd/storage/filesystem"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/canonical/lxd/shared/revert"
+	"github.com/canonical/lxd/shared/units"
+	"github.com/canonical/lxd/shared/validate"
+)
+
+// factorGiB divides a byte size value into Gibibytes.
+const factorGiB = 1024 * 1024 * 1024
+
+// CreateVolume creates an empty volume and can optionally fill it by executing the supplied filler function.
+func (d *powerflex) CreateVolume(vol Volume, filler *VolumeFiller, op *operations.Operation) error {
+	revert := revert.New()
+	defer revert.Fail()
+
+	// Get raw the volume size in GiB.
+	// PowerFlex accepts values without unit only.
+	sizeBytes, err := units.ParseByteSizeString(vol.ConfigSize())
+	if err != nil {
+		return err
+	}
+
+	sizeGiB := sizeBytes / factorGiB
+
+	client := d.client()
+	pool, err := d.resolvePool()
+	if err != nil {
+		return err
+	}
+
+	volName, err := d.getVolumeName(vol)
+	if err != nil {
+		return err
+	}
+
+	id, err := client.createVolume(volName, sizeGiB, d.getVolumeType(vol), pool.ID)
+	if err != nil {
+		return err
+	}
+
+	revert.Add(func() { _ = client.deleteVolume(id, "ONLY_ME") })
```

Comment:
whats this ONLY_ME? Needs a comment

Reply:
It tells that PowerFlex should only remove the volume and none of its child snapshots. It's explained in the docstring of `client.deleteVolume()` but I'll also add a comment here.